framework/include/utils/MooseTypes.h

```diff
@@ -121,6 +121,14 @@ template <typename>
 class ADMaterialProperty;
 class InputParameters;
 
+enum class MaterialPropState
+{
+  CURRENT = 0x1,
+  OLD = 0x2,
+  OLDER = 0x4
+};
+using MaterialPropStateInt = std::underlying_type<MaterialPropState>::type;
```

Comment:
Never noticed `underlying_type`

Reply:
it is the correct way to get an integer quantity from an enum